src/charm.py

```diff
@@ -193,15 +205,25 @@ def _get_hydra_endpoint_info(self) -> str:
                 logger.info("No hydra-endpoint-info relation data found")
         return hydra_url
 
-    def _get_kratos_endpoint_info(self, endpoint_type: str) -> str:
-        kratos_url = ""
-        if self.model.relations[KRATOS_ENDPOINTS_INTEGRATION_NAME]:
+    def _get_kratos_info(self, key: str) -> str:
+        kratos_key = ""
+        if self.kratos_info.is_ready():
+            try:
+                kratos_info = self.kratos_info.get_kratos_info()
+                kratos_key = kratos_info[key]
+            except KratosInfoRelationDataMissingError:
+                logger.info("No kratos-info relation data found")
+        return kratos_key
```

Comment:
I changed this method, it returns an empty dict in case the relation data is not found, since I want to execute a dict `get` method. Does that sound ok to you?